scale/src/commonMain/kotlin/Nice.kt

```diff
@@ -0,0 +1,28 @@
+package com.juul.krayon.scale
+
+import kotlin.math.ceil
+import kotlin.math.floor
+
+/** See equivalent [d3 function](https://github.com/d3/d3-array#nice). */
+public fun nice(start: Float, stop: Float, count: Int): List<Float> =
+    nice(start.toDouble(), stop.toDouble(), count).map { it.toFloat() }
+
+/** See equivalent [d3 function](https://github.com/d3/d3-array#nice). */
+public fun nice(start: Double, stop: Double, count: Int): List<Double> {
+    var nextStart = start
+    var nextStop = stop
+    var previousIncrement: Int? = null
+    while (true) {
+        val increment = tickIncrement(nextStart, nextStop, count)
+        if (increment == previousIncrement || increment == 0) {
+            return listOf(nextStart, nextStop)
+        } else if (increment > 0) {
+            nextStart = floor(nextStart / increment) * increment
+            nextStop = ceil(nextStop / increment) * increment
+        } else {
+            nextStart = ceil(nextStart * increment) / increment
+            nextStop = floor(nextStop * increment) / increment
+        }
+        previousIncrement = increment
+    }
```

Comment:
nit: particular reason for choosing `increment` over `step` (as used in D3)?

```suggestion
    var previousStep: Int? = null
    while (true) {
        val step = tickIncrement(nextStart, nextStop, count)
        if (step == previousStep || step == 0) {
            return listOf(nextStart, nextStop)
        } else if (increment > 0) {
            nextStart = floor(nextStart / step) * step
            nextStop = ceil(nextStop / step) * step
        } else {
            nextStart = ceil(nextStart * step) / step
            nextStop = floor(nextStop * step) / step
        }
        previousStep = step
    }
```

Reply:
`tickStep` is another, separate function that d3/krayon have. Considering the value comes from `tickIncrement` and not `tickStep`, I consider this one of the _many_ readability improvements from translating the source.